### PR TITLE
Add RRULE parsing

### PR DESCRIPTION
**Pull request by ericdude4**
From the ICalendar spec: https://icalendar.org/iCalendar-RFC-5545/3-3-10-recurrence-rule.html

This adds the ability to generate event recurrences which appear within a time range.

### CODE REVIEW

**Comment by ericdude4** [comment]
@lpil Does using the Steam module the way I have done here, immediately followed by Enum.to_list() defeat the purpose of trying to keep memory constant?

**Comment by lpil** [comment]
Yes, that would use the same amount of memory and slightly more CPU. In order to keep the amount of memory constant we need to not evaluate the stream to a list.

**Comment by ericdude4** [comment]
@lpil yea I assumed so. I actually haven't done a ton of work with the Stream module until now, but it's really cool! The get_recurrences function now returns a stream :)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # Changelog
 
+## v1.0.4 - 2020-09-04
+
+- inclue RRULE recurrence rule when parsing events.
```
Comment (lpil) [inline CHANGELOG.md]: Typo here!

**Review by lpil — APPROVED** [review]
This is great, thanks!

Besides one typo I think we're good to go.

**Comment by lpil** [comment]
Wonderful! Thank you for this, it's good to have this in.

I'll release a new version immediately

**Comment by ericdude4** [comment]
@lpil do you mind adding the `hacktoberfest-accepted` label to this pull request?

**Comment by lpil** [comment]
Does that look right @ericdude4 ?

**Comment by ericdude4** [comment]
As far as I can tell; seems they are changing the rules as they go 🤷‍♂️  Thanks!

**Pull request merged**